cdep/src/main/java/io/cdep/cdep/resolver/GithubStyleMultipackageUrlCoordinateResolver.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package io.cdep.cdep.resolver;
+
+import io.cdep.annotations.NotNull;
+import io.cdep.annotations.Nullable;
+import io.cdep.cdep.Coordinate;
+import io.cdep.cdep.Version;
+import io.cdep.cdep.utils.GithubUtils;
+import io.cdep.cdep.yml.cdep.SoftNameDependency;
+import io.cdep.cdep.yml.cdepmanifest.CDepManifestYml;
+
+import java.io.IOException;
+import java.net.URL;
+import java.security.NoSuchAlgorithmException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.cdep.cdep.utils.Invariant.require;
+
+
+/**
+ * This resolver allows you to reference a github release directly by it's URL. For example,
+ *
+ * - compile: https://github.com/google/cdep/releases/download/firebase-2.1.3-rev5/cdep-manifest-database.yml
+ *
+ * Where that URL corresponds to a coordinate like:
+ *
+ * com.github.google.cdep:firebase/database:2.1.3-rev5
+ *
+ */
+public class GithubStyleMultipackageUrlCoordinateResolver extends CoordinateResolver {
+  final private Pattern pattern = Pattern.compile("^https://(.*)/(.*)/(.*)/releases/download/(.*?)@(.*)/cdep-manifest(.*).yml$");
```

Comment:
I want to reserve this part of the github.com URL namespace for manifests that have coordinates that match the URL. That is, I don't want to allow a URL that looks like it points to SQLite but actually contains Boost.